workflow/models.py

```diff
@@ -160,7 +160,17 @@ class Meta:
         verbose_name_plural = "Organizations"
         app_label = 'workflow'
 
+    def clean_fields(self, exclude=None):
+        super(Organization, self).clean_fields(exclude=exclude)
+        if self.oauth_domains:
+            if Organization.objects.filter(
+                    oauth_domains__overlap=self.oauth_domains
```

Comment:
I'm not sure but I think you can combine these two `if` conditional.

Reply:
I thought about that. Still I am not sure but if array is none or empty and if I combine them then it will make connection for empty array. I think adding one more check is better than hitting database.

Reply:
Good point!